src/services/docker/docker.go

```diff
@@ -183,3 +183,24 @@ func (ds DockerService) Push() error {
 
 	return logger.PrintStream(pushResponse)
 }
+
+func (ds DockerService) GetImageDigest() (string, error) {
+	logger.PrintInfo("Getting manifest digest for " + ds.dockerImage.RemoteTag())
+	logger.PrintInfo("User: " + ds.AuthConfig.Username)
```

Comment:
You might want to check if the user is set here, since this is needed when pulling from a private registry

Reply:
I also just realised that this will fail if we pull from a private registry since we dont have any `registry-user` `registry-password` parameter in the deploy command.

https://github.com/nearform/initium-cli/pull/117/files#diff-53781244989afadf4bf128fdfd32e6bb60771477c410da74e6ace9dc1c5e8ae9R35

Reply:
looking at `kn service update` behaviour it looks like they always create a new revision, should probably do the same here or even use the kn client library itself if we find an easy way to integrate it.

Reply:
it looks like they are using an annotation to do this https://github.com/knative/client/pull/1364
I'm sorry @zaremba-tomasz but I'm thinking of dropping this and use the annotation approach :( 

[Here](https://github.com/knative/client/blob/main/pkg/serving/config_changes.go#L129) how to setup the annotation, we should probably use the same annotation to be compatible and eventually try to use their library for this.

Reply:
Actually @zaremba-tomasz I was thinking that having a SHA of the commit that generated the change would be way more useful, the issue is when you run the CLI from your local machine and maybe you have uncommitted changes, at the point the value could be `lastCommitSha-dirty`? https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitglossary.html#def_dirty

Reply:
> This is getting to complicated, are you sure that a new revision will not pull the new image?

The initial issue was that, the subsequent push to the repository did not create new revision at all (most probably because manifest did not change).

Reply:
> Actually @zaremba-tomasz I was thinking that having a SHA of the commit that generated the change would be way more useful, the issue is when you run the CLI from your local machine and maybe you have uncommitted changes, at the point the value could be `lastCommitSha-dirty`? https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitglossary.html#def_dirty

The issue was not related to the local environment only; I can easily reproduce this in GHA as well.

Reply:
> The initial issue was that, the subsequent push to the repository did not create new revision at all (most probably because manifest did not change).

Yes, it looks like that changing the annotation value will also trigger a new revision that will pull the image again,